utils/httpconnector/connector.go

```diff
@@ -0,0 +1,35 @@
+package httpconnector
+
+import (
+	"bytes"
+	"encoding/json"
+	"net/http"
+	"time"
+
+	"github.com/rudderlabs/rudder-server/config"
+)
+
+// MakeHTTPPostRequest makes HTTP Post call to the url with the body as payload and returns the error if any.
+// This uses basic authentication
+func MakeHTTPPostRequest(url string, payload []byte) error {
+	rawJSON, err := json.Marshal(payload)
+	if err != nil {
+		return err
+	}
+	client := &http.Client{Timeout: config.GetDuration("HttpClient.timeout", 30, time.Second)}
+	req, err := http.NewRequest("POST", url, bytes.NewBuffer(rawJSON))
```

Comment:
Be aware, that you are creating an HTTP client for every call. In our use case, it is not a problem since we send it only once, but in high traffic scenarios, it would be problematic.

My suggestion is to use a struct and save the client to the struct.

This comes to my second point, your http.Client is not testable. Please check https://golang.cafe/blog/golang-httptest-example.html